recipes/gperf/all/test_package/conanfile.py

```diff
@@ -3,6 +3,7 @@
 
 class TestPackageConan(ConanFile):
     settings = "os", "arch"
+    test_type = "build_requires"
```

Comment:
I think you can fix it by running the test in a host environment: adding `run_environment=True` to `self.run`.

Reply:
I'll give it a try.
What's wrong with my fix anyways?

Reply:
It works. But please explain your rationale to me so I don't screw up again in the future :)

Reply:
When you do a cross build, conan will build gperf once for the host machine and once for the build machine.
A build for the build machine, just to please a test seemed silly to me.
I think adding `test_type = "build_requires"` is only really needed when a recipe packages a combination of executable/generator+ library.

Reply:
So by declaring `test_type = "build_requires"` I was making conan build this package twice???

Reply:
Only when you are cross building.
Try re-adding `test_type = "build_requires"` and do a cross build. (e.g. 64-bit to 32-bit):

```
conan remove gperf -f
conan create . gperf/3.1@ -pr:h default32 -pr:b default64
```
Where default32 is for x86 (containing CFLAGS/CXXFLAGS=-m32) and default64 is for x86_64 (containing CFLAGS/CXXFLAGS=-m64)